src/topicmodeller/topicmodeller/topicmodeller.py

```diff
@@ -0,0 +1,155 @@
+        self.batch_size = batch_size
+
+    def __iter__(self):
+        scraper_documents = []
+        documents = []
+        for file_name in os.listdir(self.folder):
+
+            file_content = open(os.path.join(self.folder, file_name)).read()
+            scraper_document = _decode(file_content)
+            readable_document = _readable_document(scraper_document.html_content)
+
+            scraper_documents.append(scraper_document)
+            documents.append(_clean(_word_tokenize(readable_document)))
+
+            if len(documents) == self.batch_size:
+                yield (scraper_documents, documents)
+                scraper_documents = []
+                documents = []
+        if len(documents) != 0:
+            yield (scraper_documents, documents)
+
+
+class TopicModeller(object):
+    def __init__(self, model_data_folder, topics):
+        self.model_data_folder = model_data_folder
+
+        self.dictionary_file = os.path.join(model_data_folder, 'dictionary.dic')
+        self.lda_model_file = os.path.join(model_data_folder, 'lda.mod')
+
+        self.topics = topics
+
+        self.lda = None
+
+        if os.path.isfile(self.dictionary_file):
+            self.dictionary = corpora.Dictionary.load(self.dictionary_file)
+            # dictionary.values() is a list, looking into a list is an O(n) operation.
+            # To avoid poor performances on __filter_document method, dictionary words
+            # are "cached" on a set (O(1)).
+            self.dictionary_words = set(self.dictionary.values())
+
+        if os.path.isfile(self.lda_model_file):
+            self.lda = models.LdaModel.load(self.lda_model_file)
+
+    def initialize_dictionary(self, batched_documents):
+        self.dictionary = corpora.Dictionary()
+
+        for (_, documents) in batched_documents:
+            corpora.Dictionary.add_documents(self.dictionary, documents)
+        self.dictionary.save(self.dictionary_file)
+
+        self.dictionary_words = set(self.dictionary.values())
+
+        self.lda = None
+
+    def feed(self, batched_documents):
+        for (_, documents) in batched_documents:
+            # Construct the Corpus : Transform each document to a vector [(word_id, word_count) | word_count > 0]
+            corpus = [self.dictionary.doc2bow(self.__filter_document(document)) for document in documents]
+            self.__update_model(corpus)
+        self.lda.save(self.lda_model_file)
+
+    def classify(self, document):
+        return self.lda[self.dictionary.doc2bow(document)]
+
+    def __update_model(self, new_corpus):
+        if self.lda is None:
+            self.lda = models.LdaModel(new_corpus, id2word=self.dictionary, num_topics=self.topics)
+        else:
+            self.lda.update(new_corpus)
+
+    def __filter_document(self, document):
+        # We want to keep only known words (those on our dictionary)
+        return [word for word in document if word in self.dictionary_words]
+
```

Comment:
the standard is only one leading underscore for "private" (to be able to test it)
https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables